Ical.Net/DataTypes/CalDateTime.cs

```diff
@@ -174,271 +237,301 @@ public override int GetHashCode()
+    /// If the value is set to <see langword="null"/>, <see cref="Value"/> will be <see cref="DateTimeKind.Unspecified"/>,
+    /// and the system's local timezone will be used.
+    /// <para/>
+    /// Setting the <see cref="TzId"/> will initialize in the same way aw with the <seealso cref="CalDateTime(DateTime, string, bool)"/>.<br/>
+    /// To convert to another timezone, use <see cref="ToTimeZone"/>.
     /// </summary>
-    public string TzId
+    public string? TzId
     {
-        get
-        {
-            if (string.IsNullOrWhiteSpace(_tzId))
-            {
-                _tzId = Parameters.Get("TZID");
-            }
-            return _tzId;
-        }
+        get => _tzId;
         set
         {
             if (string.Equals(_tzId, value, StringComparison.OrdinalIgnoreCase))
             {
                 return;
             }
 
-            _asUtc = DateTime.MinValue;
-
-            var isEmpty = string.IsNullOrWhiteSpace(value);
-            if (isEmpty)
-            {
-                Parameters.Remove("TZID");
-                _tzId = null;
-                Value = DateTime.SpecifyKind(Value, DateTimeKind.Local);
-                return;
-            }
-
-            var kind = string.Equals(value, "UTC", StringComparison.OrdinalIgnoreCase)
-                ? DateTimeKind.Utc
-                : DateTimeKind.Local;
-
-            Value = DateTime.SpecifyKind(Value, kind);
-            Parameters.Set("TZID", value);
-            _tzId = value;
+            Initialize(_dateOnly, HasTime ? _timeOnly : null, value);
         }
     }
 
-    public string TimeZoneName => TzId;
+    /// <summary>
+    /// Gets the timezone name, if it references a timezone.
+    /// This is an alias for <see cref="TzId"/>.
+    /// </summary>
+    public string? TimeZoneName => TzId;
 
+    /// <inheritdoc cref="DateTime.Year"/>
     public int Year => Value.Year;
 
+    /// <inheritdoc cref="DateTime.Month"/>
     public int Month => Value.Month;
 
+    /// <inheritdoc cref="DateTime.Day"/>
     public int Day => Value.Day;
 
+    /// <inheritdoc cref="DateTime.Hour"/>
     public int Hour => Value.Hour;
 
+    /// <inheritdoc cref="DateTime.Minute"/>
     public int Minute => Value.Minute;
 
+    /// <inheritdoc cref="DateTime.Second"/>
     public int Second => Value.Second;
 
+    /// <inheritdoc cref="DateTime.Millisecond"/>
     public int Millisecond => Value.Millisecond;
 
+    /// <inheritdoc cref="DateTime.Ticks"/>
     public long Ticks => Value.Ticks;
 
+    /// <inheritdoc cref="DateTime.DayOfWeek"/>
     public DayOfWeek DayOfWeek => Value.DayOfWeek;
 
+    /// <inheritdoc cref="DateTime.DayOfYear"/>
     public int DayOfYear => Value.DayOfYear;
 
+    /// <inheritdoc cref="DateTime.Date"/>
     public DateTime Date => Value.Date;
 
+    /// <inheritdoc cref="DateTime.TimeOfDay"/>
     public TimeSpan TimeOfDay => Value.TimeOfDay;
```

Comment:
Isn't this obsolete, now that we have the `TimeOnlyValue`?

Reply:
see task in https://github.com/ical-org/ical.net/issues/646